dicoogle/src/main/java/pt/ua/dicoogle/server/users/UsersStruct.java

```diff
@@ -45,94 +44,102 @@ public static synchronized UsersStruct getInstance() {
     }
 
 
+    private UsersStruct() {
+        reset();
+
+        usersXML = new UsersXML();
+        List<User> userList = usersXML.getXML();
+
+        for (User user : userList) {
+            users.put(user.getUsername(), user);
+        }
 
-    private UsersStruct(){
-       reset();
     }
 
-    
+
     /**
      * Insert one default user
      * Username: "dicoogle"
-     * Password: "DCpassword" (hashed)
-     *
+     * Password: "dicoogle" (hashed)
+     * <p>
      * This user is administrator
      */
-    public void setDefaults(){
-        //DebugManager.getSettings().debug("Setting default user settings");
-
+    public static Collection<User> getDefaults() {
         String username = "dicoogle";
         boolean admin = true;
         String passPlainText = "dicoogle";
 
-        String passHash = HashService.getSHA1Hash(passPlainText);             //password Hash
-        String Hash = HashService.getSHA1Hash(username + admin + passHash);   //user Hash
+        //String passHash = HashService.getSHA1Hash(passPlainText);             //password Hash
 
-        users = new HashMap<String, User>();
-        users.put("dicoogle", new User(username, Hash, admin));
+        Collection<User> collection = new LinkedList<>();
```

Comment:
There are significantly better ways to make a collection with a single element. Try this instead.

```java
return Collections.singleton(new User(username, passPlainText, admin);
```

Reply:
Nice improvement. Will replace right away. Thank you!